examples/guides/mp-restful-webservice/README.adoc

```diff
@@ -669,223 +499,210 @@ accepting it. But omitting that validation lets us create an easy health check
 +
 --
-[source,java]
-// _include::0-5:{health-resource-class}[tag=readyMethod]
-----
-    @SuppressWarnings("checkstyle:designforextension")
-    @Path("/ready")
-    @GET
-    public Response ready() {
-        return Response.ok().build();
-    }
-----
---
-
-=== Update `GreetApplication`
-In the `getClasses` method add `HealthResource.class` to the arguments
-passed to `CollectionsHelper.setOf`:
+The health framework invokes `call` to retrieve the health information associated
+with this health resource.
 
 [source,java]
-// _include::0-6:{greet-application-class}[tags=getClassesMethod]
-----
-    @Override
-    public Set<Class<?>> getClasses() {
-        return CollectionsHelper.setOf(
-                GreetResource.class
-                , HealthResource.class
-        );
+// _include::0-10:{health-resource-class}[tag=callMethod,indent=0]
+----
+public HealthCheckResponse call() {
+    HealthCheckResponseBuilder builder = HealthCheckResponse.builder()
+            .name("greetingAlive"); //<1>
+    if (greeting == null || greeting.getMessage().trim().length() == 0) { //<2>
+        builder.down() //<3>
+               .withData("greeting", "not set or is empty");
+    } else {
+        builder.up(); //<4>
     }
+    return builder.build(); //<5>
+}
 ----
+<1> Create the health check with the name `greetingAlive`.
+<2> The service is alive as long as the greeting message (trimmed) is not empty.
+<3> If not, report that this liveness check is `down` and add an explanatory
+message.
+<4> If so, report that this liveness check is `up`.
+<5> Either way, build the response and return it.
+--
 
-// _include::0-47:{se-guide-adoc}[tags=rebuildAndRerunService;tryReadiness;tryLiveness]
+// _include::0-67:{se-guide-adoc}[tags=rebuildAndRerunService;tryReadiness;!se-HealthChecks-notes]
 === Stop, rebuild and rerun your service
 
 . Stop any running instance of your app.
 . Rebuild the app and then run it.
 
 
-=== Try the readiness check
-Access the readiness check endpoint:
+=== Check the server's health
+Run this command
 [source,bash]
-curl -i -X GET http://localhost:8080/ready
+curl -i -X GET http://localhost:8080/health
+
+and you should see output like this
+[listing,subs=+quotes]
+----
+{"outcome":"UP","checks":[{"name":"deadlock","state":"UP"},{"name":"diskSpace","state":"UP","data":{"free":"180.64 GB","freeBytes":193955860480,"percentFree":"38.79%","total":"465.72 GB","totalBytes":500068036608}},*{"name":"greetingAlive","state":"UP"}*,{"name":"heapMemory","state":"UP","data":{"free":"230.86 MB","freeBytes":242074232,"max":"4.00 GB","maxBytes":4294967296,"percentFree":"99.39%","total":"256.00 MB","totalBytes":268435456}}]}
+----
 
-Our readiness check returns no payload, just the 200 status, so you won't see any data
-displayed. The `-i` option shows the
-200 status in the response.
+The first item is `outcome` which describes the overall health of the
+server based on all the other indicators. All indicators are `UP` so the outcome
```

Comment:
I suggest changing the sentence, "All indicators are `UP`..."

to

"The `status` of all the indicators is `UP` in this example. So the `outcome` field shows `UP`.

Reply:
done